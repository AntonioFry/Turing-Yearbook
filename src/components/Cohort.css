.cohort {
  width: 100%;
}

.people {
  display: grid;
  grid-template-columns: repeat(3, 1fr);
  grid-gap: 20px;
  padding: 20px;
}

h2 {
  background-color: rgba(255, 255, 255, 0.95);
}