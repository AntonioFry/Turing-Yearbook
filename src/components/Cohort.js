import React from 'react';
import Person from './Person';
import './Cohort.css';

const Cohort = (props) => {
  console.log(props);
  const cards = props.people.map(person => {
    return <Person key={props.id} image={person.photo} name={person.name}
    quote={person.quote} superlative={person.superlative} />
  })
  
  return (
    <main>
      <h2>Cohort</h2>
        {cards}
    </main>
  )
}

export default Cohort;
