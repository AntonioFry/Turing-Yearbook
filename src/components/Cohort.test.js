import React from 'react';
import { shallow, mount } from 'enzyme';
import Cohort from './Cohort';

describe('Cohort', () => {

  var props;
  beforeEach(function() {
    props = {
      people: [
        { id: 1, name: 'name', photo: 'image', quote: 'quote', superlative: 'superlative' },
        { id: 2, name: 'name', photo: 'image', quote: 'quote', superlative: 'superlative' }
      ]
    }
  });

  it('match the snapshot with all the data passed in correctly', () => {
    const wrapper = shallow(
    <Cohort
      people={props.people}
    />)

    expect(wrapper).toMatchSnapshot();
  });
})