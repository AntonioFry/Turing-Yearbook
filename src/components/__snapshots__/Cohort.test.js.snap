// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cohort match the snapshot with all the data passed in correctly 1`] = `
<section
  className="cohort"
>
  <h2>
    Cohort
  </h2>
  <section
    className="people"
  >
    <Person
      key="1"
      name="name"
      photo="image"
      quote="quote"
      superlative="superlative"
    />
    <Person
      key="2"
      name="name"
      photo="image"
      quote="quote"
      superlative="superlative"
    />
  </section>
</section>
`;
